server/channels/app/notification_push.go

```diff
@@ -143,9 +143,12 @@ func (a *App) sendPushNotification(notification *PostNotification, user *model.U
 	cancelled := false
 	a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
 		rejectionReason := hooks.NotificationWillBePushed(&model.PluginPushNotification{
-			Post:    notification.Post.ForPlugin(),
-			Channel: notification.Channel,
-			UserID:  user.Id,
+			Post:               notification.Post.ForPlugin(),
+			Channel:            notification.Channel,
+			UserID:             user.Id,
+			ExplicitMention:    explicitMention,
+			ChannelWideMention: channelWideMention,
+			ReplyToThreadType:  replyToThreadType,
```

Comment:
For the plugin caller, can we document these on `PluginPushNotification` to explain when to set them?

Reply:
Done, thanks!